src/geometries/primitives/parametrizedcurve.jl

```diff
@@ -0,0 +1,48 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ParametrizedCurve(func, ab = (0.0, 1.0))
+
+A parametrized curve is a curve defined by a function `func` that maps a parameter `t` to a `Point` in space.
+The parameter `t` is defined in the interval `[a, b]`. The curve can only be evaluated for `t` in the
+interval `[0, 1]`.
+
+## Examples
+
+```julia
+ParametrizedCurve(t -> Point(cospi(2t), sinpi(2t)))
+```
+"""
+struct ParametrizedCurve{M<:Meshes.Manifold,C<:Meshes.CRS,T<:Real,F<:Function} <: Primitive{M,C}
+  a::T
+  b::T
+  func::F
```

Comment:
Let's store the fields `func::F` and `range::R` with `F<:Function` and `R<:Tuple` in that order, to match the constructor.

Reply:
Done in [e830105](https://github.com/JuliaGeometry/Meshes.jl/pull/1093/commits/e830105ee365337755550969f06c79166b245051).